In [1]:
import pandas as pd
import numpy as np

df = pd.read_csv('processed_ExcelQuizMhs.csv')

df.head()  

name               P1  \
0  TIM PENGAMPU PA/TA FIK, -TUGAS AKHIR 2 - A11.4701  [192, 99, 4, 0]   
1  EGIA ROSI SUBHIYAKTO, M.Kom-SOFTWARE QUALITY A...   [43, 27, 1, 0]   
2  EGIA ROSI SUBHIYAKTO, M.Kom-REKAYASA PERANGKAT...   [27, 13, 0, 0]   
3  EGIA ROSI SUBHIYAKTO, M.Kom-SOFTWARE QUALITY A...    [27, 8, 1, 0]   
4  EGIA ROSI SUBHIYAKTO, M.Kom-REKAYASA PERANGKAT...    [31, 7, 0, 0]   

                 P2                P3               P4                P5  \
0  [182, 109, 4, 0]  [186, 106, 2, 1]  [222, 70, 3, 0]  [191, 100, 4, 0]   
1    [44, 26, 1, 0]    [45, 26, 0, 0]   [53, 17, 0, 1]    [44, 26, 0, 1]   
2    [26, 14, 0, 0]    [24, 16, 0, 0]    [35, 5, 0, 0]    [27, 13, 0, 0]   
3    [26, 10, 0, 0]    [26, 10, 0, 0]    [33, 3, 0, 0]     [27, 9, 0, 0]   
4     [32, 6, 0, 0]     [33, 5, 0, 0]    [34, 4, 0, 0]     [31, 7, 0, 0]   

                 P6                P7                P8                P9  \
0  [192, 100, 3, 0]  [184, 106, 5, 0]  [190, 100, 5, 0]  [186, 106, 3, 0]   
1    [44, 26, 0, 1]    [42, 28, 1, 0]    [45, 25, 1, 0]    [43, 28, 0, 0]   
2    [23, 17, 0, 0]    [24, 16, 0, 0]    [25, 15, 0, 0]    [24, 16, 0, 0]   
3     [29, 7, 0, 0]    [24, 12, 0, 0]     [28, 8, 0, 0]     [27, 9, 0, 0]   
4     [29, 9, 0, 0]    [28, 10, 0, 0]     [30, 8, 0, 0]     [32, 6, 0, 0]   

                P10  knowledge mastery  teaching skill  teacher skill  
0  [188, 103, 4, 0]              87.41               0          88.25  
1    [46, 25, 0, 0]              87.04               0          87.23  
2    [24, 16, 0, 0]              88.00               0          88.50  
3    [25, 11, 0, 0]              90.74               0          92.78  
4     [29, 9, 0, 0]              94.04               0          93.51

In [2]:
# remove column 'knowledge mastery'
df = df.drop(['knowledge mastery'], axis=1)
df = df.drop(['teaching skill'], axis=1)
df = df.drop(['teacher skill'], axis=1)

df.head()

name               P1  \
0  TIM PENGAMPU PA/TA FIK, -TUGAS AKHIR 2 - A11.4701  [192, 99, 4, 0]   
1  EGIA ROSI SUBHIYAKTO, M.Kom-SOFTWARE QUALITY A...   [43, 27, 1, 0]   
2  EGIA ROSI SUBHIYAKTO, M.Kom-REKAYASA PERANGKAT...   [27, 13, 0, 0]   
3  EGIA ROSI SUBHIYAKTO, M.Kom-SOFTWARE QUALITY A...    [27, 8, 1, 0]   
4  EGIA ROSI SUBHIYAKTO, M.Kom-REKAYASA PERANGKAT...    [31, 7, 0, 0]   

                 P2                P3               P4                P5  \
0  [182, 109, 4, 0]  [186, 106, 2, 1]  [222, 70, 3, 0]  [191, 100, 4, 0]   
1    [44, 26, 1, 0]    [45, 26, 0, 0]   [53, 17, 0, 1]    [44, 26, 0, 1]   
2    [26, 14, 0, 0]    [24, 16, 0, 0]    [35, 5, 0, 0]    [27, 13, 0, 0]   
3    [26, 10, 0, 0]    [26, 10, 0, 0]    [33, 3, 0, 0]     [27, 9, 0, 0]   
4     [32, 6, 0, 0]     [33, 5, 0, 0]    [34, 4, 0, 0]     [31, 7, 0, 0]   

                 P6                P7                P8                P9  \
0  [192, 100, 3, 0]  [184, 106, 5, 0]  [190, 100, 5, 0]  [186, 106, 3, 0]   
1    [44, 26, 0, 1]    [42, 28, 1, 0]    [45, 25, 1, 0]    [43, 28, 0, 0]   
2    [23, 17, 0, 0]    [24, 16, 0, 0]    [25, 15, 0, 0]    [24, 16, 0, 0]   
3     [29, 7, 0, 0]    [24, 12, 0, 0]     [28, 8, 0, 0]     [27, 9, 0, 0]   
4     [29, 9, 0, 0]    [28, 10, 0, 0]     [30, 8, 0, 0]     [32, 6, 0, 0]   

                P10  
0  [188, 103, 4, 0]  
1    [46, 25, 0, 0]  
2    [24, 16, 0, 0]  
3    [25, 11, 0, 0]  
4     [29, 9, 0, 0]

Criteria Weighs :
- SS = 0.4
- KS = 0.3
- TS = 0.2
- STS = 0.1

Alternative P1 :
- SS = 192
- KS = 99
- TS = 4
- STS = 0

Alternative P2 :
- SS = 182
- KS = 109
- TS = 4
- STS = 0

Alternative P3 :
- SS = 186
- KS = 106
- TS = 2
- STS = 1

Alternative P5 :
- SS = 191
- KS = 100
- TS = 4
- STS = 0

Alternative P10 :
- SS = 188
- KS = 103
- TS = 4
- STS = 0

Min-Max Normalized Ratings :

SS = (max-current)/(max-min)  

p1-SS = (192 - 192) / (192 - 182)
P1-SS = 0 / 10
p1-SS = 0

p2-SS = (192 - 182) / (192 - 182)
p2-SS = 10 / 10
p2-SS = 1

P3-SS = (192 - 186) / (192 - 182)
P3-SS = 6 / 10
P3-SS = 0.6

P5-SS = (192 - 191) / (192 - 182)
P5-SS = 1 / 10
P5-SS = 0.1

P10-SS = (192 - 188) / (192 - 182)
P10-SS = 4 / 10
P10-SS = 0.4

p1-KS = (109 - 99) / (109 - 99)
P1-KS = 10 / 10
p1-KS = 1

P2-KS = (109 - 109) / (109 - 99)
P2-KS = 0 / 10
P2-KS = 0

P3-KS = (109 - 106) / (109 - 99)
P3-KS = 3 / 10
P3-KS = 0.3

P5-KS = (109 - 100) / (109 - 99)
P5-KS = 9 / 10
P5-KS = 0.1

P10-KS = (109 - 103) / (109 - 99)
P10-KS = 6/10
P10-KS = 0.6

P1-TS = (4-4)/(4-2)
P1-TS = 0

P2-TS = (4-4)/(4-2)
P2-TS = 0

P3-TS = (4-2)/(4-2)
P3-TS = 1

P5-TS = (4-4)/(4-2)
P5-TS = 0

P10-TS = (4-4)/(4-2)
P10-TS = 0

P1-STS = 0

P2-STS = 0

P3-STS = 1

P5-STS = 0

P10-STS = 0

Weighted Sums :

P1
SS = 0.4 * 0 = 0  
KS = 0.3 * 1 = 0.3
TS = 0.2 * 0 = 0  
STS = 0.1 * 0 = 0  


P2
SS = 0.4 * 1 = 0.4  
KS = 0.3 * 0 = 0  
TS = 0.2 * 0 = 0  
STS = 0.1 * 0 = 0  


P3
SS = 0.4 * 0.6 = 0.24  
KS = 0.3 * 0.3 = 0.09  
TS = 0.2 * 1 = 0.2  
STS = 0.1 * 1 = 0.1  


P5
SS = 0.4 * 0.1 = 0.04  
KS = 0.3 * 0.1 = 0.03  
TS = 0.2 * 0 = 0  
STS = 0.1 * 0 = 0  


P10
SS = 0.4 * 0.4 = 0.16  
KS = 0.3 * 0.6 = 0.18  
TS = 0.2 * 0 = 0  
STS = 0.1 * 0 = 0  

P1 total = 0.3  
P2 total = 0.4  
P3 total = 0.63  
P5 total = 0.07  
P10 total = 0.34  

P3 is the best decision for calculating knowledge mastery aspect
????????